### PR TITLE
(restructure) - deprecate ie11 support

### PR DESCRIPTION
**Pull request by JoviDeCroock**

### CODE REVIEW

**Comment by developit** [comment]
Interesting this didn't improve benchmarks more - I wonder if we should be jumping straight to restructuring?

I was thinking it might be worth switching the transpiler config to use targets.esmodules rather than last 2 versions.

File: babel.config.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ module.exports = function(api) {
 					modules: noModules ? false : 'auto',
 					exclude: ['@babel/plugin-transform-typeof-symbol'],
 					targets: {
-						browsers: ['last 2 versions', 'IE >= 9']
+						browsers: ['last 2 versions', 'not IE 11']
```
Comment (marvinhagemeister) [inline babel.config.js:28]: Apparently `not IE 11` includes IE 10, blackberry 10 and other relics from the past: https://browserslist.dev/?q=bGFzdCAyIHZlcnNpb25zLCBub3QgSUUgMTE%3D

Not sure what a good replacement value would be that only includes somewhat modern browsers. Maybe @developit knows more?

Comment (developit) [reply babel.config.js]: Need to include 'not dead'. But this was the main reason I am thinking we might be better off using targets.esmodules.

Comment (JoviDeCroock) [reply babel.config.js]: esbuild seems to fail when I do that 😅

Comment (developit) [reply babel.config.js]: Probably a simplified browserslist implementation. This should work and is equivalent:

```
"Firefox>=60, chrome>=61, and_chr>=61, Safari>=10.1, iOS>=10.3, edge>=16, opera>=48, op_mob>=48, Samsung>=8.2, not dead"
```

Comment (JoviDeCroock) [reply babel.config.js]: I don't think microbundle actually looks at this babel.config 😅

Comment (developit) [reply babel.config.js]: correct - it uses `"browserslist"` from the package.json.

**Comment by developit** [comment]
Weird, coverage never ran on the merge commit.

**Comment by JoviDeCroock** [comment]
Yes coverage seems to be acting up on `main` and my other PR as well

**Comment by JoviDeCroock** [comment]
I think there's still a few suites not running

**Comment by developit** [comment]
@danielweck heh - after I took that screenshot, someone from the V8 team pointed out that the `for..of` test was incorrectly referencing a global variable instead of the loop variable. Now things look a bit more like what you posted (notably with Safari suffering from quite poor `for..of` performance).

**Comment by fzzle** [comment]
Chrome 90:
image

**Comment by gustavpursche** [comment]
Firefox 90.0a1 (2021-05-20) on Ubuntu:

Screenshot from 2021-05-24 15-59-25

**Comment by developit** [comment]
Nice! Thanks for the cross-platform checks @fzzle & @gustavpursche! I think they confirm things nicely - we likely can't make generalized recommendation between "unrolled" vs "unrolled X2", but the fact that those two are consistently either nearly-fastest (Chrome/Edge ~92) or significantly faster (Firefox & Safari) means we should be leaning _slightly_ towards unrolling where it can be done without sacrificing other metrics (size, simplicity, etc). FWIW I think the Edge result is a bit of an anomaly, and m93 will likely exhibit performance slightly more in line with m90 (a very small advantage for unrolling).

File: src/create-context.js [hunk 2]
```diff
@@ -40,7 +40,9 @@ export function createContext(defaultValue, contextId) {
 						// 	c.context[contextId] = _props.value;
 						// 	enqueueRender(c);
 						// });
-						subs.some(enqueueRender);
+						for (let sub of subs) {
```
Comment (developit) [inline src/create-context.js:43]: lingering for..of here

File: src/diff/component.js [hunk 3]
```diff
@@ -62,7 +62,7 @@ export function renderComponent(
 		c = internal._component;
 	} else {
 		// Instantiate the new component
-		if ('prototype' in type && type.prototype.render) {
+		if (internal._flags & TYPE_CLASS) {
```
Comment (andrewiggins) [inline src/diff/component.js:65]: Could we leave this line as-is for now? I have another PR where the `TYPE_CLASS` flag gets removed as part of the "make our internals more agnostic to component implementation" work

Comment (developit) [reply src/diff/component.js:65]: Definitely - I couldn't generate any meaningful metrics to support this change when I tried, best to just leave it as-is.

**Pull request merged**